test_arm.c:58:73: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘int64_t’

It's better to use `PRIuX` formatters
```
test_arm.c: In function ‘print_insn_detail’:
test_arm.c:58:73: warning: format ‘%x’ expects argument of type ‘unsigned int’, but argument 3 has type ‘int64_t’ {aka ‘long int’} [-Wformat=]
   58 |                                 printf("\t\toperands[%u].type: IMM = 0x%x\n", i, op->imm);
      |                                                                        ~^        ~~~~~~~
      |                                                                         |          |
      |                                                                         |          int64_t {aka long int}
      |                                                                         unsigned int
      |                                                                        %lx
test_arm.c:85:73: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 3 has type ‘int64_t’ {aka ‘long int’} [-Wformat=]
   85 |                                 printf("\t\toperands[%u].type: P-IMM = %u\n", i, op->imm);
      |                                                                        ~^        ~~~~~~~
      |                                                                         |          |
      |                                                                         |          int64_t {aka long int}
      |                                                                         unsigned int
      |                                                                        %lu
test_arm.c:88:73: warning: format ‘%u’ expects argument of type ‘unsigned int’, but argument 3 has type ‘int64_t’ {aka ‘long int’} [-Wformat=]
   88 |                                 printf("\t\toperands[%u].type: C-IMM = %u\n", i, op->imm);
      |                                                                        ~^        ~~~~~~~
      |                                                                         |          |
      |                                                                         |          int64_t {aka long int}
      |                                                                         unsigned int
      |                                                                        %lu
```

--- a/cstool/cstool_arm.c
+++ b/cstool/cstool_arm.c
@@ -28,14 +28,12 @@ void print_insn_detail_arm(csh handle, cs_insn *ins)
 			case ARM_OP_REG:
 				printf("\t\toperands[%u].type: REG = %s\n", i, cs_reg_name(handle, op->reg));
 				break;
-			case ARM_OP_IMM: {
-				bool neg_imm = op->imm < 0;
-				if (neg_imm)
-					printf("\t\toperands[%u].type: IMM = -0x%lx\n", i, -(op->imm));
+			case ARM_OP_IMM:
+				if (op->imm < 0)
+					printf("\t\toperands[%u].type: IMM = -0x%" PRIx64 "\n", i, -(op->imm));
 				else
-					printf("\t\toperands[%u].type: IMM = 0x%lx\n", i, op->imm);
+					printf("\t\toperands[%u].type: IMM = 0x%" PRIx64 "\n", i, op->imm);
 				break;
-			}
 			case ARM_OP_PRED:
 				printf("\t\toperands[%u].type: PRED = %d\n", i, op->pred);
 				break;
@@ -64,10 +62,10 @@ void print_insn_detail_arm(csh handle, cs_insn *ins)
 
 				break;
 			case ARM_OP_PIMM:
-				printf("\t\toperands[%u].type: P-IMM = %u\n", i, op->imm);
+				printf("\t\toperands[%u].type: P-IMM = %" PRIu64 "\n", i, op->imm);
 				break;
 			case ARM_OP_CIMM:
-				printf("\t\toperands[%u].type: C-IMM = %u\n", i, op->imm);
+				printf("\t\toperands[%u].type: C-IMM = %" PRIu64 "\n", i, op->imm);
 				break;
 			case ARM_OP_SETEND:
 				printf("\t\toperands[%u].type: SETEND = %s\n", i, op->setend == ARM_SETEND_BE? "be" : "le");

--- a/tests/test_arm.c
+++ b/tests/test_arm.c
@@ -55,7 +55,10 @@ static void print_insn_detail(csh cs_handle, cs_insn *ins)
 				printf("\t\toperands[%u].type: REG = %s\n", i, cs_reg_name(cs_handle, op->reg));
 				break;
 			case ARM_OP_IMM:
-				printf("\t\toperands[%u].type: IMM = 0x%x\n", i, op->imm);
+				if (op->imm < 0)
+					printf("\t\toperands[%u].type: IMM = -0x%" PRIx64 "\n", i, -(op->imm));
+				else
+					printf("\t\toperands[%u].type: IMM = 0x%" PRIx64 "\n", i, op->imm);
 				break;
 			case ARM_OP_FP:
 #if defined(_KERNEL_MODE)
@@ -82,10 +85,10 @@ static void print_insn_detail(csh cs_handle, cs_insn *ins)
 
 				break;
 			case ARM_OP_PIMM:
-				printf("\t\toperands[%u].type: P-IMM = %u\n", i, op->imm);
+				printf("\t\toperands[%u].type: P-IMM = %" PRIu64 "\n", i, op->imm);
 				break;
 			case ARM_OP_CIMM:
-				printf("\t\toperands[%u].type: C-IMM = %u\n", i, op->imm);
+				printf("\t\toperands[%u].type: C-IMM = %" PRIu64 "\n", i, op->imm);
 				break;
 			case ARM_OP_SETEND:
 				printf("\t\toperands[%u].type: SETEND = %s\n", i, op->setend == ARM_SETEND_BE? "be" : "le");

